### ISSUE TITLE
Cell sorts not used in sort inference?

### ISSUE THREAD

**Issue by bmmoore** [kframework/k-legacy#1691]
In this definition, the variable `I` is given sort `K`, instead of inferring sort `Int` from the sort of the `<count>` cell.

```
require "domains.k"

module TEST-SYNTAX
 imports TEST-COMMON
 imports ID
endmodule

module TEST-COMMON
 imports DOMAINS-COMMON

 syntax Pgm ::= "start"
endmodule

module TEST
 imports TEST-COMMON
 imports DOMAINS

 configuration <k>$PGM:K</k><count>0::Int</count>

 rule start => 1
 rule <k>`I => .K` ...</k><count>_ => I</count>
endmodule
```

@radumereuta, I'll look into this, but if it's an issue with the loose cell grammar you are generating for parsing you should take it over.

**Comment by bmmoore** [comment]
I see the RULE_CELLS case in RuleGrammarGenerator indeed ignores the sort of leaf cells, but I think I can fix it.

**Comment by bmmoore** [comment]
Fixed in #1703
